modules/ding_reservation/ding_reservation.module

```diff
@@ -490,25 +490,28 @@ function ding_reservation_reservations_form($form, &$form_state, $reservations,
   }
 
   foreach ($reservations as $reservation) {
-    /** @var \TingEntity $entity */
-    $entity = $reservation->entity;
-    if (!is_object($entity) || NULL === $entity->getTingObject()) {
-      $entity = ding_provider_get_pseudo_entity($reservation->ding_entity_id);
-      // Don't link the title as the data well do not known this title. It may
-      // be a inter library loan.
-      $title = (!empty($entity) && $entity->getTitle()) ? $entity->getTitle() : $reservation->display_name;
-    }
-    else {
-      // Create title that links to the object.
+    $title = t('Title not available');
+    $creators = $mat_type = $cover = NULL;
+
+    // If the provider was able to produce an entity we'll use that since we can
+    // show more information and link to the entity.
+    if (is_object($reservation->entity)) {
+      $entity = $reservation->entity;
+
       $uri = entity_uri('ting_object', $entity);
       $title = l($entity->getTitle(), $uri['path']);
+      $creators = $entity->getCreators();
+      $mat_type = $entity->getType();
+      $cover = field_view_field('ting_object', $entity, 'ting_cover', 'user_list');
+    }
+    // Else the provider has the option to supply a display name and author.
+    elseif (!empty($reservation->display_name)) {
+      $title = $reservation->display_name;
+      $creators = $reservation->author;
+    }
+    else {
```

Comment:
Is this else needed will it not default to this from line 493?

Reply:
I missed that one. It's not needed. Will remove.